x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/create_trusted_app_flyout.tsx

```diff
@@ -120,15 +120,34 @@ export const CreateTrustedAppFlyout = memo<CreateTrustedAppFlyoutProps>(
       'trustedAppsByPolicyEnabled'
     );
 
+    const [wasByPolicy, setWasByPolicy] = useState(!isGlobalEffectScope(formValues.effectScope));
+    // set initial state of `wasByPolicy` that checks if the initial state of the exception was by policy or not
+    useEffect(() => {
+      if (!isFormDirty && formValues.effectScope) {
+        setWasByPolicy(!isGlobalEffectScope(formValues.effectScope));
+      }
+    }, [isFormDirty, formValues.effectScope]);
+
     const isGlobal = useMemo(() => {
       return isGlobalEffectScope((formValues as NewTrustedApp).effectScope);
     }, [formValues]);
 
     const showExpiredLicenseBanner = useMemo(() => {
       return (
-        isTrustedAppsByPolicyEnabled && !isPlatinumPlus && isEditMode && (!isGlobal || isFormDirty)
+        isTrustedAppsByPolicyEnabled &&
+        !isPlatinumPlus &&
+        isEditMode &&
+        wasByPolicy &&
```

Comment:
Fixes an error: the banner is displayed when editing the form even if it wasn't by policy

Reply:
no need to use `useMemo` here.